### PR TITLE
Keep performance CI jobs' CSV output as build artifacts.

### PR DESCRIPTION
**Pull request by hidmic**
Precisely what the title says. Depends on https://github.com/ros-infrastructure/ros_buildfarm/pull/784. Addresses https://github.com/ros2/buildfarm_perf_tests/issues/30.

### CODE REVIEW

**Comment by hidmic** [comment]
@cottsay does this align with what you had in mind?

**Comment by hidmic** [comment]
@cottsay friendly ping.

File: foxy/ci-nightly-performance.yaml [hunk 1]
```diff
@@ -105,6 +105,11 @@ targets:
 type: ci-build
 underlay_from_ci_jobs:
 - nightly-extra-rmw-release
+keep_files:
```
Comment (cottsay) [inline foxy/ci-nightly-performance.yaml]:
```suggestion
archive_files:
```

**Review by cottsay — APPROVED** [review]

File: foxy/ci-nightly-performance.yaml (hunk 1 above)
Comment (hidmic) [reply foxy/ci-nightly-performance.yaml]: Done in https://github.com/ros2/ros_buildfarm_config/pull/92/commits/de5100df0fa9e4461bfd3fbf80386c45df4c6dae

File: foxy/ci-nightly-performance.yaml [hunk 2]
```diff
@@ -105,6 +105,11 @@ targets:
 type: ci-build
 underlay_from_ci_jobs:
 - nightly-extra-rmw-release
+archive_files:
+  - ws/test_results/buildfarm_perf_tests/performance_test_results_*.csv
+  - ws/test_results/buildfarm_perf_tests/overhead_test_results_*.csv
+  - ws/test_results/buildfarm_perf_tests/overhead_node_test_results_*.csv
+  - ws/test_results/buildfarm_perf_tests/performance_test_two_process_results_*.csv
```
Comment (cottsay) [inline foxy/ci-nightly-performance.yaml]: For consistency, this YAML file isn't indenting for lists.

```suggestion
- ws/test_results/buildfarm_perf_tests/performance_test_results_*.csv
- ws/test_results/buildfarm_perf_tests/overhead_test_results_*.csv
- ws/test_results/buildfarm_perf_tests/overhead_node_test_results_*.csv
- ws/test_results/buildfarm_perf_tests/performance_test_two_process_results_*.csv
```

**Review by cottsay — APPROVED** [review]
One more quick change.

File: foxy/ci-nightly-performance.yaml (hunk 2 above)
Comment (hidmic) [reply foxy/ci-nightly-performance.yaml]: Good point, see 216f99ac9872f0dbed497629eaeec9829d5f1ea0.

**Pull request merged**